design/survival/design_survival.qmd

```diff
@@ -0,0 +1,1967 @@
+---
+title: "Design Doc for Survival Analysis with RobinCar2"
+author: Daniel Sabanes Bove
+date: "`r format(Sys.Date(), '%d %b %Y')`"
+date-format: long
+bibliography: references.bib
+number-sections: true
+---
+
+This design doc shall serve to align on the design of the survival analysis implementation in RobinCar2.
+
+# User interface
+
+The proposal for the user interface is the function call of the form:
+
+```{r}
+#| eval: false
+
+robin_surv(
+  Surv(time, event) ~ treatment * s1 + covariates,
+  data = df,
+  treatment = pb(s1),
```

Comment:
how the model knows that the variable `treatment` is the treatment variable, if not provided in the treatment argument?